roles/openshift_sanitize_inventory/tasks/deprecations.yml

```diff
@@ -1,21 +1,23 @@
 ---
 
 - name: Check for usage of deprecated variables
-  set_fact:
-    __deprecation_message: "{{ __deprecation_message | default( __deprecation_header ) }} \n\t{{ item }}"
-  when:
-  - hostvars[inventory_hostname][item] is defined
-  with_items: "{{ __warn_deprecated_vars }}"
+  check_deprecated:
+    facts: "{{ hostvars[inventory_hostname] }}"
+    vars: "{{ __warn_deprecated_vars }}"
+    header: "{{ __deprecation_header }}"
+  register: dep_check
+  run_once: true
```

Comment:
This will result in the variables only being checked on one host (most likely first master).  This will lead to a situation where other host groups that define variables (or even using hostvars) could still have deprecated variables.